internal/renderers/skia/lib.rs

```diff
@@ -60,42 +64,48 @@ pub struct SkiaRenderer {
     path_cache: ItemCache<Option<(Vector2D<f32, PhysicalPx>, skia_safe::Path)>>,
     rendering_metrics_collector: RefCell<Option<Rc<RenderingMetricsCollector>>>,
     rendering_first_time: Cell<bool>,
-    surface: DefaultSurface,
+    surface: once_cell::unsync::OnceCell<Box<dyn Surface>>,
+    window_handle_provider: Rc<dyn raw_window_handle::HasWindowHandle>,
+    display_handle_provider: Rc<dyn raw_window_handle::HasDisplayHandle>,
 }
 
 impl SkiaRenderer {
     /// Creates a new renderer is associated with the provided window adapter.
     pub fn new(
-        window_handle: raw_window_handle::WindowHandle<'_>,
-        display_handle: raw_window_handle::DisplayHandle<'_>,
-        size: PhysicalWindowSize,
+        window_handle_provider: Rc<dyn raw_window_handle::HasWindowHandle>,
+        display_handle_provider: Rc<dyn raw_window_handle::HasDisplayHandle>,
```

Comment:
From an API point of view, this is not ideal.   This always force the handle i a Rc.  which may not be easy.
I think using a `raw_window_handle::WindowHandle<'_>`  is the right API.

Is it worth trying to solve this problem? what's wrong with passing the size?

Could this be done with a `init(handle, handle)`  instead?

Reply:
> Is it worth trying to solve this problem?

That's a good question.

On the one hand it doesn't make sense to initialise a renderer with a given handle to a native window but say: "Hey, the size is not known yet". Of course the native window has a size, it must have - I think.

On the other hand the size can be zero initially, which doesn't work here. And the Rc<> here helps to ensure that the window handle remains valid throughout the lifetime - this really adds safety to the API. And thirdly, our API doesn't allow for the renderer to be created lazily (because of the renderer() getter on the WindowAdapter).

In conclusion, I'll close this PR and we stick to the existing API. If we later find that we can also do this conveniently without the size then we can still add an overload to the C++ SkiaRenderer API.

At least we know that the current approach definitely works for the users we know of and tried this with.